src/datasources/queues/queue-consumer.service.interface.ts

```diff
@@ -0,0 +1,15 @@
+import { ConsumeMessage } from 'amqplib';
+
+export const IQueueConsumerService = Symbol('IQueueConsumerService');
+
+export interface IQueueConsumerService {
+  /**
+   * Checks if the configured queue consumer is connected.
+   */
+  isReady(): boolean;
+
+  subscribe(
+    queueName: string,
+    fn: (msg: ConsumeMessage) => Promise<void>,
```

Comment:
The function `fn` would be called every time a message arrives to the queue named `queueName`.